### DISCUSSION TITLE
What happens to modifiers during redeclarations?

### DISCUSSION THREAD

**Discussion by casella** [OpenModelica/OpenModelica#11256] [Q&A]
Consider this simple MWE:
```modelica
package TestRedeclare
  model M1
    parameter Real p = 1;
  end M1;
  model S1
    replaceable M1 m(p = 10);
  end S1;
  model M2
    extends M1;
    parameter Real q = 4;
  end M2;
  model S2
    extends S1(redeclare M2 m);
  end S2;
end TestRedeclare;
```
When flattening S2 with OMC I get
```modelica
class TestRedeclare.S2
  parameter Real m.p = 10.0;
  parameter Real m.q = 4.0;
end TestRedeclare.S2;
```
Apparently, the modifier `p = 10` which appears in the definition of `m` in `S1` is retained when `m` is redeclared. This makes sense, it would be annoying if one had to repeat all the bindings every time a component is redeclared.

However, we failed to identify where in the MLS this is explained. In [Section 7.3](https://specification.modelica.org/maint/3.6/inheritance-modification-and-redeclaration.html#redeclarationhttps://specification.modelica.org/maint/3.6/inheritance-modification-and-redeclaration.html#redeclaration) one reads: 

> In redeclarations some parts of the original declaration is automatically inherited by the new declaration. This is intended to make it easier to write declarations by not having to repeat common parts of the declarations, and does in particular apply to prefixes that must be identical. The inheritance only applies to the declaration itself and not to elements of the declaration.

What is not clear is what are the "some parts" that are automatically retained, and which other parts are not retained, and based on what crieteria. @perost, is this clear to you? Should we suggest to improve the text of the specification?

Keeping @AndreaBartolini in the loop.

**Comment by perost** [comment]
Which parts are retained is explained immediately following the quoted text, so that part is clear. But modifiers are not part of this but instead part of the constraining type mechanism covered by [7.3.2](https://specification.modelica.org/maint/3.6/inheritance-modification-and-redeclaration.html#constraining-type) (which still applies here even though you're not explicitly using `constrainedby`).

**Comment by casella** [comment]
> But modifiers are not part of this but instead part of the constraining type mechanism covered by [7.3.2](https://specification.modelica.org/maint/3.6/inheritance-modification-and-redeclaration.html#constraining-type) (which still applies here even though you're not explicitly using `constrainedby`).

OK, we missed that. Intuitively it makes sense, but we wanted to make sure there were no ambiguities involved. Thanks!
